lib/iris/tests/unit/mesh/components/test_MeshCoord.py

```diff
@@ -1083,3 +1083,17 @@ def test_standard_name(self):
         for c in self.mesh.coords(axis="y"):
             c.standard_name = "grid_latitude"
         assert self.meshcoord.standard_name == self.coord_on_mesh.standard_name
+
+    def test_updates_only_once(self, mocker):
+        mocked = mocker.patch.object(MeshCoord, "_load_points_and_bounds")
+        mocked.return_value = (
+            np.zeros(
+                3,
+            ),
+            np.zeros((3, 3)),
+        )
+        self.coord_on_mesh.points = np.zeros(3)
+        # Only the first update should actually update
+        _ = self.meshcoord.update_from_mesh()
+        _ = self.meshcoord.update_from_mesh()
+        mocked.assert_called_once()
```

Comment:
It's probably worth writing a similar test to check that it's called twice when expected.

Reply:
It should never be called twice. If it's updated once, that means it works as expected. i.e. it's called the first time when timestamps are expected to be different, it's not called the second time when timestamps are expected to be aligned.

Reply:
Ok I think this has now been addressed 👍